src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+        assert out_dir.is_dir()
+
+        video_path = pathlib.Path(self.args["video_path"])
+        assert video_path.is_file()
+
+        exp_id = video_path.name.split("_")[0]
+
+        roi_path = pathlib.Path(self.args["roi_path"])
+        assert roi_path.is_file()
+
+        graph_path = pathlib.Path(self.args["graph_path"])
+        assert graph_path.is_file()
+
+        proj = self.get_max_avg(video_path)
+        self.logger.info("Calculated mean and max images...")
+        avg_img = proj["avg"]
+        max_img = proj["max"]
+        corr_img = graph_to_img(graph_path)
+        self.logger.info("Calculated correlation image...")
+
+        quantiles = (self.args["low_quantile"], self.args["high_quantile"])
+        q0, q1 = np.quantile(max_img, quantiles)
+        max_img = normalize_array(array=max_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(avg_img, quantiles)
+        avg_img = normalize_array(array=avg_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(corr_img, quantiles)
+        corr_img = normalize_array(array=corr_img,
+                                   lower_cutoff=q0,
+                                   upper_cutoff=q1)
+
+        self.logger.info("Normalized images...")
+
+        with open(roi_path, "rb") as in_file:
+            extract_roi_list = sanitize_extract_roi_list(
+                json.load(in_file))
+
+        self.logger.info("Creating and writing ROI artifacts...")
+        for roi in extract_roi_list:
+            self.write_thumbnails(extract_roi=roi,
+                                  max_img=max_img,
+                                  avg_img=avg_img,
+                                  corr_img=corr_img,
+                                  exp_id=exp_id,
+                                  out_dir=out_dir)
+
+        self.logger.info(f"Created ROI artifacts in {time.time()-t0:.0f} "
+                         "seconds.")
+
+    def get_max_avg(self,
+                    video_path: pathlib.Path) -> Dict[str, np.ndarray]:
+        """Compute the mean and max projection from a movie on disk.
+
+        Parameters
+        ----------
+        video_path : pathlib.Path
+            Path to HDF5 file containing full movie. Movie is assume to be
+            stored in a dataset named "data".
+
+        Returns
+        -------
+        projections : Dict[str, np.ndarray]
+            Dictionary of image projects of the movie.
+
+            "mean" : Mean projection of the movie.
+            "max" : Max projection of the movie.
+        """
+        with h5py.File(video_path, "r") as in_file:
+            video_data = in_file["data"][()]
+        max_img = np.max(video_data, axis=0)
+        avg_img = np.mean(video_data, axis=0)
+
+        return {"max": max_img, "avg": avg_img}
+
+    def write_thumbnails(self,
```

Comment:
Similarly, I don't think this needs to be a class member. It shouldn't use `self`.

Reply:
I'll change it up so that the `Path` isn't input and other config args are used directly rather than pass them into the method. I don't think that there's a use for this method outside of this pipeline so I'll make it more internal.

Reply:
Sounds good to me